### DISCUSSION TITLE
How to find Android Toolbar in NavigationPage?

### DISCUSSION THREAD

**Discussion by SergTomcat** [dotnet/maui#14490] [Q&A] [answered]
I'm trying to modify toolbar in NavigationPage. Cannot find the control.

I tried this code:

```
#if ANDROID
	//var vw = MyMainActivity.Instance;
	var vw = Platform.CurrentActivity;

	var toolbar1 = vw.FindViewById<AndroidX.AppCompat.Widget.Toolbar>(Resource.Id.toolbar);
	var toolbar2 = vw.FindViewById<AndroidX.AppCompat.Widget.Toolbar>(Resource.Layout.Toolbar);

	var toolbarx = vw.FindViewById<Android.Widget.Toolbar>(Resource.Id.toolbar);
	var toolbary = vw.FindViewById<Android.Widget.Toolbar>(Resource.Layout.Toolbar);
						
	var toolbara = vw.FindViewById(Resource.Id.toolbar);
	var toolbarb = vw.FindViewById(Resource.Layout.Toolbar);

#endif
```

but everything return null.

Please, advise how to get toolbar control (or any control in general)

**Comment by mhrastegari** [comment]
I hope this will help

https://learn.microsoft.com/en-us/dotnet/maui/user-interface/pages/navigationpage?view=net-maui-7.0#display-views-in-the-navigation-bar

**Comment by SergTomcat** [comment]
@mhrastegari , thank you for reply, I know about TitleView and, actually, using it, but I need to tweak toolbar items on android (overflow icon especially: [this one](https://learn.microsoft.com/en-us/dotnet/api/android.widget.toolbar.overflowicon?view=xamarin-android-sdk-13)). Need to find this toolbar.

**Comment by pauldiston** [comment]
@SergTomcat Have you managed to find the toolbar for a NavigationPage?

**Reply by SergTomcat** [reply]
No, still no success.

**Reply by pauldiston** [reply, marked as answer]
@SergTomcat The following thread might be of interest to you - [https://stackoverflow.com/questions/75596420/how-do-i-add-a-listener-to-the-android-toolbar-in-maui/76056039#76056039](https://stackoverflow.com/questions/75596420/how-do-i-add-a-listener-to-the-android-toolbar-in-maui/76056039#76056039)

**Reply by SergTomcat** [reply]
Thank you! "navigationlayout_appbar" did the trick, finally got my Toolbar :)
